File: assets/icon/generate_icon.py
#!/usr/bin/env python3
"""
Generate CoachGuru app icon (1024x1024 PNG) for Google Play Store
"""

from PIL import Image, ImageDraw, ImageFont
import math

def create_coachguru_icon():
    # Create 1024x1024 image
    size = 1024
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Create gradient background (simplified - using solid color for now)
    # In a real implementation, you'd create a proper gradient
    draw.rectangle([0, 0, size, size], fill=(11, 45, 92, 255))  # #0B2D5C
    
    # Add gradient effect by drawing diagonal overlay
    for i in range(size):
        alpha = int(255 * (i / size))
        color = (14, 95, 216, alpha)  # #0E5FD8 with varying alpha
        draw.line([(0, i), (i, 0)], fill=color)
    
    # Draw CG monogram
    try:
        # Try to use a system font
        font = ImageFont.truetype("/System/Library/Fonts/Arial.ttf", 200)
    except:
        try:
            font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", 200)
        except:
            font = ImageFont.load_default()
    
    # Draw white CG text
    text = "CG"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    x = (size - text_width) // 2
    y = (size - text_height) // 2
    draw.text((x, y), text, fill=(255, 255, 255, 255), font=font)
    
    # Draw orange tactic arrow
    arrow_color = (255, 176, 0, 255)  # #FFB000
    
    # Arrow path
    arrow_points = [
        (700, 300), (800, 300), (780, 280), (820, 320),
        (780, 360), (800, 340), (700, 340)
    ]
    draw.polygon(arrow_points, fill=arrow_color)
    
    # Arrow circle
    draw.ellipse([735, 305, 765, 335], fill=arrow_color)
    
    # Save the image
    img.save('coachguru_app_icon_1024.png', 'PNG')
    print("CoachGuru app icon generated: coachguru_app_icon_1024.png")

if __name__ == "__main__":
    create_coachguru_icon()

File: assets/icon/generate_icon.js
// Simple script to generate CoachGuru app icon
// This creates a canvas-based PNG of the app icon

const fs = require('fs');
const { createCanvas } = require('canvas');

// Create a 1024x1024 canvas
const canvas = createCanvas(1024, 1024);
const ctx = canvas.getContext('2d');

// Create gradient background
const gradient = ctx.createLinearGradient(0, 0, 1024, 1024);
gradient.addColorStop(0, '#0B2D5C');
gradient.addColorStop(1, '#0E5FD8');

// Fill background with gradient
ctx.fillStyle = gradient;
ctx.fillRect(0, 0, 1024, 1024);

// Add rounded corners effect (simplified)
ctx.globalCompositeOperation = 'destination-in';
ctx.beginPath();
ctx.roundRect(0, 0, 1024, 1024, 200);
ctx.fill();

// Reset composite operation
ctx.globalCompositeOperation = 'source-over';

// Draw CG monogram
ctx.fillStyle = '#FFFFFF';
ctx.font = 'bold 200px Arial';
ctx.textAlign = 'center';
ctx.textBaseline = 'middle';
ctx.fillText('CG', 512, 512);

// Draw orange tactic arrow
ctx.fillStyle = '#FFB000';
ctx.beginPath();
ctx.moveTo(700, 300);
ctx.lineTo(800, 300);
ctx.lineTo(780, 280);
ctx.lineTo(820, 320);
ctx.lineTo(780, 360);
ctx.lineTo(800, 340);
ctx.lineTo(700, 340);
ctx.closePath();
ctx.fill();

// Add circle to arrow
ctx.beginPath();
ctx.arc(750, 320, 15, 0, 2 * Math.PI);
ctx.fill();

// Save as PNG
const buffer = canvas.toBuffer('image/png');
fs.writeFileSync('coachguru_app_icon_1024.png', buffer);

console.log('CoachGuru app icon generated: coachguru_app_icon_1024.png');
